proxy/coco/src/peer/run_state.rs

```diff
@@ -29,12 +34,108 @@ const DEFAULT_SYNC_PERIOD: Duration = Duration::from_secs(5);
 pub enum Command {
     /// Start the announcement subroutine.
     Announce,
+    Request(RequestCommand),
     /// Initiate a full sync with `PeerId`.
     SyncPeer(PeerId),
     /// Start sync timeout.
     StartSyncTimeout(Duration),
 }
 
+impl From<RequestCommand> for Command {
+    fn from(other: RequestCommand) -> Self {
+        Self::Request(other)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum RequestCommand {
+    Clone(RadUrl),
+    Found(RadUrl),
+    Query(RadUrn),
+}
+
+impl RequestCommand {
+    pub async fn run<W>(self, state: Lock, waiting_room: W)
```

Comment:
I see where you going with this. Yet it feels out of place to have this in the run_state module where we so far have been exclusively be concerned with the state transition of the peer event machine. It should follow the organisation of the other module like announcement and sync.